spec/lang/step/operators.md

```diff
@@ -250,6 +250,54 @@ impl<M: Memory> Machine<M> {
 }
 ```
 
+### Raw Equality
+```rust
+impl<M: Memory> Machine<M> {
+    fn load_raw_data(&mut self, ptr : Pointer<<M as Memory>::Provenance>, ptr_ty : PtrType) -> Result<List<u8>> {
+        let PtrType::Ref { mutbl, pointee } = ptr_ty else {
+            panic!("non-reference input to raw equality");
+        };
+        if mutbl == Mutability::Mutable {
+            panic!("mutable reference to raw equality");
+        }
+        let Layout { size, align, ..} = pointee;
+        let bytes = self.mem.load(ptr, size, align, Atomicity::None)?;
+        Ok(bytes.map(|byte| {
+            let Some(data) = byte.data() else { panic!("byte is uninitialized"); };
+            if let Some(_) = byte.provenance() { panic!("byte has provenance"); }
```

Comment:
These should not panic. A panic indicates a bug in the interpreter, but actual user programs can call raw_eq on uninit bytes or those with provenance. It is UB to do so, but that's a bug in the program, not the interpreter.